### PR TITLE
update the SRL demo.

### LINKED ISSUE #1788: Semantic Role Labeling Demo only provide v2 trainer, but not provide model inference code.
Related issue #1785.

The paddle [srl demo](https://github.com/PaddlePaddle/Paddle/tree/develop/demo/semantic_role_labeling) only provide v2 API trainer, which is `api_train_v2.py`. The `predict.py` is not related to this trainer, and the `train.sh`, `data` cannot be used due to missing files.

### LINKED ISSUE #2080: The inference speed of V2 API is much slower than the V1 version
A user reported that the speed of inference by using V2 APIs is slower than before, as I know, there are some not quite good designs. Is it possible to optimized?

Besides, we do not provide an official doc on how to do batch inference. I want to know if there are some rules to follow or some things I should pay attentions to.

### PR DESCRIPTION
**Pull request by lcy-seso**
fix #1788. fix #2080.
1. add v2 inferring to SRL demo.
2. inferring over the entire batch.
3. inferring by load the parameter only once to speed up.

### CODE REVIEW

**Comment by lcy-seso** [comment]
- I use model saved by the first pass and test the entire test set which contains 5267 samples.
- To avoid loading the parameters in every tested batch, the inferring time is reduced from 96.5s to 78.19s.

File: demo/semantic_role_labeling/api_train_v2.py [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
 import numpy as np
-import paddle.v2 as paddle
+import gzip
+import logging
 import paddle.v2.dataset.conll05 as conll05
+import paddle.v2.evaluator as evaluator
+import paddle.v2 as paddle
 
+logger = logging.getLogger('paddle')
+logger.setLevel(logging.WARN)
```
Comment (qingqing01) [inline demo/semantic_role_labeling/api_train_v2.py]: 下面的print是不是也可以统一用logger?

File: demo/semantic_role_labeling/api_train_v2.py [hunk 2]
```diff
@@ -179,12 +194,60 @@ def event_handler(event):
...
+
+        if isinstance(event, paddle.event.EndPass):
+            # save parameters
+            with gzip.open('params_pass_%d.tar.gz' % event.pass_id, 'w') as f:
+                parameters.to_tar(f)
+
+            result = trainer.test(reader=reader, feeding=feeding)
+            print "\nTest with Pass %d, %s" % (event.pass_id, result.metrics)
+
+    if not is_predict:
```
Comment (qingqing01) [inline demo/semantic_role_labeling/api_train_v2.py]: if else分支，建议改成两个函数train和infer。`trainer = paddle.trainer.SGD` 和 `reader`等也不属于infer分支~

File: demo/semantic_role_labeling/api_train_v2.py (hunk 1 above)
Comment (lcy-seso) [inline demo/semantic_role_labeling/api_train_v2.py]: done.

File: demo/semantic_role_labeling/api_train_v2.py (hunk 2 above)
Comment (lcy-seso) [inline demo/semantic_role_labeling/api_train_v2.py]: done.

**Pull request merged**